### PR TITLE
[Packetbeat] Update DNS protocol to use ECS fields

### PR DESCRIPTION
**Pull request by andrewkroh**
This updates the DNS protocol to have more closely follow ECS.
The DNS tunneling dashboard has been updated to work with the new
field names.

In order to better interoperate with other data sources the trailing dot
has been removed from domain names. For example, previously Packetbeat
would produce `dns.question.name:elastic.co.` and now it will simply produce
`dns.question.name:elastic.co`. It's a breaking change but it will be make it
easier to pivot with other data sources.

Part of #7968

Here's a summary of what fields changed.

Changed

- bytes_in -> source.bytes
- bytes_out -> destination.bytes
- notes -> error.message
- responsetime -> event.duration (unit are now nanoseconds)
- transport -> network.transport

Added

- event.end
- event.start
- network.bytes
- network.community_id
- network.protocol = dns
- network.type

Unchanged Packetbeat Fields

- method - dns opcode
- query = class {{ dns.question.class }}, type {{ dns.question.type }}, {{ dns.question.name }}
- request - text representation of the entire request
- response - text representation of the entire response
- resource - dns.question.name
- status
- type = dns (we might remove this since we have event.dataset)

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -532,6 +532,30 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
+--------------------------------------------------------------------
+Dependency: github.com/elastic/ecs
+Revision: 69de90eb6493e0804405321f48adfdfa488d6498
```
Comment (ruflin) [inline NOTICE.txt]: My first though was to fix it to Beta2 but at this stage the go code was not in yet :-(

Comment (andrewkroh) [reply NOTICE.txt]: Luckily the code generated from this commit is exactly the same as what would be in beta2 if I backported (I started a backport and realize they’d be the same).d

**Comment by andrewkroh** [comment]
The are no changes needed to fields.yml in this PR. When I finish updating the last protocol in Packetbeat then there should be several updates to removing unused fields. And then I can add aliases to ecs-migration.yml. None of the DNS specific fields were changed.

I’ll update the changelog.

**Comment by ruflin** [comment]
As far as I can see for example `bytes_out` was renamed to `destination.bytes`? That would mean `bytes_out` in the fields.yml should become an alias to `destination.bytes`? Or you can't change it yet because it's also used in other places / protocols?

For the `ecs-migration.yml`: I would prefer to have a list for each protocol so if we map `bytes_out` to `destination.bytes` multiple time, it's ok if it shows up there multiple times.

What we often do in these ECS migration PR's is also having in the PR description the list of fields which we renamed.

**Pull request merged**